plugins/inventory/foreman.py

```diff
@@ -238,10 +296,235 @@ def _get_hostvars(self, host, vars_prefix='', omitted_vars=()):
                 hostvars[vars_prefix + k] = v
         return hostvars
 
+    def _fetch_params(self):
+        options = ("no", "yes")
+        params = dict()
+        try:
+            self.want_location = self.get_option('report').get('want_location')
+        except Exception:
+            self.want_location = True
```

Comment:
Can we, instead of the whole `try/except` jumping, do something like that?

```python
report_options = {'want_location': True, 'want_organization': True, … }
report_options.update(self.get_option('report') or {})
```

`get_option()` returns `None` if the option is not set at all (so in this case we take the `or {}`) and a dict with the set options if any were set.